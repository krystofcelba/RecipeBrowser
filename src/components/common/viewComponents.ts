import { View } from 'react-native';
import namedGlamorous from './namedGlamorous';

import AppTheme from 'src/assets/appTheme';

export const PaddedView = namedGlamorous(View, 'PaddedView')({
  padding: AppTheme.defaultPadding,
});
