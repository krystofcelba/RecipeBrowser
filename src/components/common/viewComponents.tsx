import React from 'react';
import { KeyboardAvoidingView as RNKeyboardAvoidingView, Platform } from 'react-native';
import glamorous from 'glamorous-native';
import namedGlamorous from './namedGlamorous';

import AppTheme from 'src/assets/appTheme';

export const { View } = glamorous;

export const PaddedView = namedGlamorous(View, 'PaddedView')({
  padding: AppTheme.defaultPadding,
});

export const KeyboardAvoidingView = ({ children, ...otherProps }) => (
  <RNKeyboardAvoidingView
    testID="recipesListScreen"
    keyboardVerticalOffset={60}
    behavior={Platform.OS === 'ios' ? 'padding' : undefined}
    {...otherProps}
  >
    {children}
  </RNKeyboardAvoidingView>
);
