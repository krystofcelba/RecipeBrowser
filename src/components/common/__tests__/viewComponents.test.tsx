import React from 'react';
import { shallow } from 'enzyme';
import { View, PaddedView } from '../viewComponents';

describe('View components', () => {
  it('View renders as expected', () => {
    const wrapper = shallow(<View />);
    expect(wrapper).toMatchSnapshot();
  });

  it('PaddedView renders as expected', () => {
    const wrapper = shallow(<PaddedView />);
    expect(wrapper).toMatchSnapshot();
  });
});
