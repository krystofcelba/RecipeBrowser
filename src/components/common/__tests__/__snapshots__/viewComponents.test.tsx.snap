// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View components PaddedView renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 10,
      },
    ]
  }
/>
`;

exports[`View components View renders as expected 1`] = `
<View
  style={null}
/>
`;
